### DISCUSSION TITLE
how to define multiple values with same named attributed?

### DISCUSSION THREAD

**Discussion by dzmitry-lahoda** [terranix/terranix#68] [Q&A] [answered]
for example `filter`?

```hcl
data "aws_ami" "ubuntu" {
  most_recent = true

  filter {
    name   = "name"
    values = ["ubuntu/images/hvm-ssd/ubuntu-focal-20.04-amd64-server-*"]
  }

  filter {
    name   = "virtualization-type"
    values = ["hvm"]
  }

  owners = ["099720109477"] # Canonical
}
```

**Comment by dzmitry-lahoda** [comment]
```nix
  data = {
    aws_ami = {
      node = {
        most_recent = true;
        filter = [
          {
            name = "name";
            values = [ "ubuntu/images/hvm-ssd/ubuntu-focal-20.04-amd64-server-*" ];

          }
          {
            name = "virtualizaiton-type";
            values = [ "hvm" ];
          }
        ];
        owners = [ "099720109477" ]; # Canonical
      };
    };
  };
```

not found in docs. glad it seems works :)

**Comment by mrVanDalo** [comment]
Good point, ... parameters which can be used multiple times in a block must be defined as a list in terranix. Will put this in the documentation, thanks.

**Comment by terlar** [comment, marked as answer]
Since terranix is just an abstraction on the Terraform JSON Configuration syntax, this kind of information can be found here:
- https://developer.hashicorp.com/terraform/language/syntax/json

More specifically this case is nested block mapping:
- https://developer.hashicorp.com/terraform/language/syntax/json#nested-block-mapping
